raphtory/src/python/graph/pandas/loaders.rs

```diff
@@ -235,15 +242,110 @@ pub(crate) fn load_edges_from_df<'a, S: AsRef<str>>(
     Ok(())
 }
 
-pub(crate) fn load_node_props_from_df<'a>(
+pub(crate) fn load_edges_deletions_from_df<'a, const N: usize, S: AsRef<str>>(
+    df: &'a PretendDF,
+    size: usize,
+    src: &str,
+    dst: &str,
+    time: &str,
+    layer: Option<S>,
+    layer_in_df: bool,
+    graph: &InnerTemporalGraph<N>,
+) -> Result<(), GraphError> {
+    let layer = lift_layer(layer, layer_in_df, df);
+
+    if let (Some(src), Some(dst), Some(time)) = (
+        df.iter_col::<u64>(src),
+        df.iter_col::<u64>(dst),
+        df.iter_col::<i64>(time),
+    ) {
+        let triplets = src
+            .map(|i| i.copied())
+            .zip(dst.map(|i| i.copied()))
+            .zip(time);
+        for (((src, dst), time), layer) in tqdm!(
+            triplets.zip(layer),
+            desc = "Loading edges",
+            total = size,
+            animation = kdam::Animation::FillUp,
+            unit_scale = true
+        ) {
+            if let (Some(src), Some(dst), Some(time)) = (src, dst, time) {
+                graph.delete_edge(*time, src, dst, layer.as_deref());
```

Comment:
im confused, how does it know if an edge has to be deleted if you are checking if it exists time, src, dst? does that mean every edge is going to be marked as deleted?